src/sleep/sleep.c

```diff
@@ -263,40 +263,57 @@ static int execute(
 }
 
 static int execute_s2h(const SleepConfig *sleep_config) {
-        _cleanup_close_ int tfd = -1;
-        struct itimerspec ts = {};
         int r;
-
         assert(sleep_config);
+        usec_t suspend_interval = 1 * USEC_PER_HOUR;
```

Comment:
maybe initialize this from `sc->hibernate_delay_sec`, so that it can be re-purposed?

Reply:
it can be re-purposed that way but the challenge i see there is then i will not be able to update it within the sleep.c then when i need to set value on in how much time system should check back instead of a fixed time.

`if (estimated_total_discharge_time < 1)`
-------->          `suspend_interval = 30 * USEC_PER_MINUTE;`

Reply:
if we do not need a different suspend intervalwhen discharge time is less than an hour, i can remove this and use hibernate_delay_sec.

Reply:
ahh. okay. Understood. Will updated.